File: src/game/item/ItemType.java
package game.item;

import game.Vector;

public enum ItemType {
	Key,
	HealthPack,
	AmmoPack;
	
	public Item instantiate(Vector pos) {
		switch (this) {
		case Key:
			return new Key(pos);
		case HealthPack:
			return new HealthPack(pos);
		case AmmoPack:
			return new AmmoPack(pos);
		}
		return null;
	}
	
	public int getCode() {
		switch (this) {
		case Key:
			return 0;
		case HealthPack:
			return 1;
		case AmmoPack:
			return 2;
		}
		return -1;
	}
	
	public static ItemType fromCode(int code) {
		switch (code) {
		case 0:
			return Key;
		case 1:
			return HealthPack;
		case 2:
			return AmmoPack;
		}
		return null;
	}
}

File: ts/game/item/ItemType.ts
import { Vector } from '../Vector.js';
import { Item } from './Item.js';
import { Key } from './Key.js';
import { HealthPack } from './HealthPack.js';
import { AmmoPack } from './AmmoPack.js';

export enum ItemType {
  Key,
  HealthPack,
  AmmoPack
}

export function instantiateItem(t: ItemType, pos: Vector): Item {
  switch (t) {
    case ItemType.Key:
      return new Key(pos);
    case ItemType.HealthPack:
      return new HealthPack(pos);
    case ItemType.AmmoPack:
      return new AmmoPack(pos);
  }
}

export function getCode(t: ItemType): number { return t; }
export function fromCode(code: number): ItemType { return code as ItemType; }
